infra/lib/connect-widget-stack.ts

```diff
@@ -9,19 +9,21 @@ import * as cloudfront_origins from "aws-cdk-lib/aws-cloudfront-origins";
 import * as iam from "aws-cdk-lib/aws-iam";
 import * as lambda from "aws-cdk-lib/aws-lambda";
 import { Construct } from "constructs";
-import { EnvConfig } from "./env-config";
+import { ConnectWidgetConfig, EnvConfig } from "./env-config";
 import { addErrorAlarmToLambdaFunc } from "./util";
 
 interface ConnectWidgetStackProps extends StackProps {
-  config: EnvConfig;
+  config: Omit<EnvConfig, "connectWidget" | "connectWidgetUrl"> & {
+    connectWidget: ConnectWidgetConfig;
+  };
```

Comment:
Is this a bug? This is omitting the keys "connectWidget" & "connectWidgetUrl", but then adding `connectWidget`?

Reply:
This is just making the resulting type to require `connectWidget`. It removes that property from the original type (where it can be present or not based on `connectWidget`), and adds it again forcing it to be present.

Its a downstream change from adjusting types so we can safely use the underlying properties without relying on `!` to disable the compiler complaining of accessing a property that might be undefined.

Reply:
BTW, this is just on type level/definition, not changing objects/memory.